main.py

```diff
@@ -55,130 +55,119 @@
-        return web.HTTPFound('/error')
-    valid_cookie = check_cookie(request)
-    if not valid_cookie:
-        return web.Response(text=login_file, content_type='text/html')
-    return web.HTTPFound('/')
-
-
-# Check the user credentials and set cookie on home page.
-async def login(request):
-    form = await request.post()
-    username = form.get('username')
-    password = form.get('password')
-    if not json_reader(CONFIG_FILE):
-        return web.HTTPFound('/error')
-    if not check_credentials(username, password):
-        return web.HTTPFound('/redirect')
-    # The user is successfully logged in, set the cookie on the index page.
-    response = web.HTTPFound('/')
-    expires = datetime.datetime.utcnow() + datetime.timedelta(days=30)
-    COOKIE_EXPIRE = expires.strftime("%a, %d %b %Y %H:%M:%S GMT")
-    response.set_cookie(COOKIE_NAME, value=COOKIE_VALUE, expires=COOKIE_EXPIRE)
-    return response
-
-
-# Routes
-app = web.Application()
-app.router.add_get('/', home)
-app.router.add_get('/log', login_page)
-app.router.add_post('/login', login)
-app.router.add_get('/redirect', redirect)
-app.router.add_get('/error', server_error)
-web.run_app(app, host=HOST, port=PORT)
+    async def server_error(self, request):
+        valid_cookie = self._check_cookie(request)
+        if not valid_cookie:
+            check_login_file = self._check_the_file(LOGIN_HTML)
+            if not check_login_file:
+                return web.Response(text=ERROR, content_type='text/html')
+            return web.HTTPFound('/log')
+        return web.HTTPFound('/')
+
+    async def home(self, request):
+        site_file = self._check_the_file(SITE_HTML)
+        if not site_file:
+            return web.HTTPFound('/error')
+        valid_cookie = self._check_cookie(request)
+        if not valid_cookie:
+            return web.HTTPFound('/log')
+        return web.Response(text=site_file, content_type='text/html')
+
+    async def redirect(self, request):
+        redirect_file = self._check_the_file(REDIRECT_HTML)
+        if not redirect_file:
+            return web.HTTPFound('/error')
+        valid_cookie = self._check_cookie(request)
+        if not valid_cookie:
+            return web.Response(text=redirect_file, content_type='text/html')
+        return web.HTTPFound('/')
+
+    async def login_page(self, request):
+        login_file = self._check_the_file(LOGIN_HTML)
+        if not login_file:
+            return web.HTTPFound('/error')
+        valid_cookie = self._check_cookie(request)
+        if not valid_cookie:
+            return web.Response(text=login_file, content_type='text/html')
+        return web.HTTPFound('/')
+
+    async def login(self, request):
+        form = await request.post()
+        self.username = form.get('username')
+        self.password = form.get('password')
+        if not self._check_the_file(CONFIG_FILE):
+            return web.HTTPFound('/error')
+        if not self._check_credentials():
```

Comment:
Why comes check_crdentials not immediately after form get calls? Why check file in between? This split the logic into two separate block for nothing! This is no novel, this is shit from a reader point of view. It makes understanding code unnecessarily complex.

And please, check_credentials should have the user and password as arguments, saving all variables as instance variables is braindead. Really, please enlight me and explain me when do you think a class variable should be used and when not